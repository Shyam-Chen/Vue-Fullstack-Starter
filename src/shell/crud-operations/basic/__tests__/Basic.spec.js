import Vuex from 'vuex';
import { createLocalVue, shallowMount } from '@vue/test-utils';

import Basic from '../Basic';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Basic', () => {
  let [wrapper, store] = [];

  beforeEach(() => {
    store = new Vuex.Store();
    wrapper = shallowMount(Basic, { store, localVue });
  });

  it('should render an initial component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
