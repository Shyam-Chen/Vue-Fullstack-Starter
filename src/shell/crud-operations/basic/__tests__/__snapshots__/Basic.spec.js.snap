// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic should render an initial component 1`] = `
<v-container-stub id="basic" tag="div" fluid="true">
  <v-layout-stub tag="div" column="">
    <div class="headline">CRUD Operations - Basic</div>
    <v-layout-stub tag="div" row="">
      <v-btn-stub color="primary" tag="button" activeclass="" type="button" class="ml-0">Add</v-btn-stub>
    </v-layout-stub>
    <v-layout-stub tag="div" row="">
      <v-card-stub loaderheight="4" tag="div">
        <v-card-title-stub class="o-card-title">
          <div class="title">Board</div>
          <v-spacer-stub></v-spacer-stub>
          <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" appendicon="search" backgroundcolor="" hidedetails="true" label="Search" loaderheight="2" clearicon="$clear" singleline="true" type="text"></v-text-field-stub>
        </v-card-title-stub>
        <v-data-table-stub items="[object Object],[object Object],[object Object],[object Object]" options="[object Object]" sortby="" sortdesc="" customsort="function sortItems(items, sortBy, sortDesc, locale, customSorters) {
          if (sortBy === null || !sortBy.length) return items;
          var stringCollator = new Intl.Collator(locale, {
            sensitivity: 'accent',
            usage: 'sort'
          });
          return items.sort(function (a, b) {
            var _a, _b;

            for (var i = 0; i < sortBy.length; i++) {
              var sortKey = sortBy[i];
              var sortA = getObjectValueByPath(a, sortKey);
              var sortB = getObjectValueByPath(b, sortKey);

              if (sortDesc[i]) {
                _a = __read([sortB, sortA], 2), sortA = _a[0], sortB = _a[1];
              }

              if (customSorters &amp;&amp; customSorters[sortKey]) {
                var customResult = customSorters[sortKey](sortA, sortB);
                if (!customResult) continue;
                return customResult;
              } // Check if both cannot be evaluated


              if (sortA === null &amp;&amp; sortB === null) {
                continue;
              }

              _b = __read([sortA, sortB].map(function (s) {
                return (s || '').toString().toLocaleLowerCase();
              }), 2), sortA = _b[0], sortB = _b[1];

              if (sortA !== sortB) {
                if (!isNaN(sortA) &amp;&amp; !isNaN(sortB)) return Number(sortA) - Number(sortB);
                return stringCollator.compare(sortA, sortB);
              }
            }

            return 0;
          });
        }" page="1" itemsperpage="10" groupby="" groupdesc="" customgroup="function groupItems(items, groupBy, groupDesc) {
          var key = groupBy[0];
          return items.reduce(function (acc, item) {
            var val = getObjectValueByPath(item, key);
            (acc[val] = acc[val] || []).push(item);
            return acc;
          }, {});
        }" locale="en-US" search="" customfilter="function defaultFilter(value, search, item) {
          return value != null &amp;&amp; search != null &amp;&amp; typeof value !== 'boolean' &amp;&amp; value.toString().toLocaleLowerCase().indexOf(search.toLocaleLowerCase()) !== -1;
        }" serveritemslength="-1" itemkey="id" value="" expanded="" mobilebreakpoint="600" noresultstext="$vuetify.dataIterator.noResultsText" nodatatext="$vuetify.noDataText" loadingtext="$vuetify.dataIterator.loadingText" selectablekey="isSelectable" headers="[object Object],[object Object],[object Object],[object Object]" expandicon="$expand" hide-actions="" select-all=""></v-data-table-stub>
      </v-card-stub>
    </v-layout-stub>
    <aside>
      <v-dialog-stub opendelay="0" closedelay="0" contentclass="" maxwidth="500px" origin="center center" retainfocus="true" transition="dialog-transition" width="auto">
        <v-card-stub loaderheight="4" tag="div">
          <v-card-title-stub>Add</v-card-title-stub>
          <v-card-text-stub>
            <v-layout-stub tag="div" row="">
              <div class="o-text-field">
                <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="Primary" loaderheight="2" clearicon="$clear" type="text" name="primary"></v-text-field-stub>
              </div>
              <div class="o-text-field">
                <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="Accent" loaderheight="2" clearicon="$clear" type="text" name="accent"></v-text-field-stub>
              </div>
            </v-layout-stub>
          </v-card-text-stub>
          <v-card-actions-stub>
            <v-btn-stub color="error" tag="button" activeclass="" type="button" flat="">Cancel</v-btn-stub>
            <v-btn-stub color="success" tag="button" activeclass="" type="button" flat="">Add</v-btn-stub>
          </v-card-actions-stub>
        </v-card-stub>
      </v-dialog-stub>
      <v-dialog-stub opendelay="0" closedelay="0" contentclass="" maxwidth="500px" origin="center center" retainfocus="true" transition="dialog-transition" width="auto">
        <v-card-stub loaderheight="4" tag="div">
          <v-card-title-stub>Edit</v-card-title-stub>
          <v-card-text-stub>
            <v-layout-stub tag="div" row="">
              <div class="o-text-field">
                <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="Primary" loaderheight="2" clearicon="$clear" type="text" name="primary"></v-text-field-stub>
              </div>
              <div class="o-text-field">
                <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="Accent" loaderheight="2" clearicon="$clear" type="text" name="accent"></v-text-field-stub>
              </div>
            </v-layout-stub>
          </v-card-text-stub>
          <v-card-actions-stub>
            <v-btn-stub color="error" tag="button" activeclass="" type="button" flat="">Cancel</v-btn-stub>
            <v-btn-stub color="success" tag="button" activeclass="" type="button" flat="">Save</v-btn-stub>
          </v-card-actions-stub>
        </v-card-stub>
      </v-dialog-stub>
      <v-dialog-stub opendelay="0" closedelay="0" contentclass="" maxwidth="500px" origin="center center" retainfocus="true" transition="dialog-transition" width="auto">
        <v-card-stub loaderheight="4" tag="div">
          <v-card-title-stub>View</v-card-title-stub>
          <v-card-text-stub>
            <v-layout-stub tag="div" row="">
              <div class="o-text-field">
                <v-text-field-stub errorcount="1" errormessages="" messages="" readonly="true" rules="" successmessages="" backgroundcolor="" label="Primary" loaderheight="2" clearicon="$clear" type="text" name="primary"></v-text-field-stub>
              </div>
              <div class="o-text-field">
                <v-text-field-stub errorcount="1" errormessages="" messages="" readonly="true" rules="" successmessages="" backgroundcolor="" label="Accent" loaderheight="2" clearicon="$clear" type="text" name="accent"></v-text-field-stub>
              </div>
            </v-layout-stub>
          </v-card-text-stub>
          <v-card-actions-stub>
            <v-btn-stub color="success" tag="button" activeclass="" type="button" flat="">Quit</v-btn-stub>
          </v-card-actions-stub>
        </v-card-stub>
      </v-dialog-stub>
      <v-dialog-stub opendelay="0" closedelay="0" contentclass="" maxwidth="500px" origin="center center" retainfocus="true" transition="dialog-transition" width="auto">
        <v-card-stub loaderheight="4" tag="div">
          <v-card-title-stub>Delete</v-card-title-stub>
          <v-card-text-stub>
            Are you sure that you want to delete it?
          </v-card-text-stub>
          <v-card-actions-stub>
            <v-btn-stub color="success" tag="button" activeclass="" type="button" flat="">Cancel</v-btn-stub>
            <v-btn-stub color="error" tag="button" activeclass="" type="button" flat="">Confirm</v-btn-stub>
          </v-card-actions-stub>
        </v-card-stub>
      </v-dialog-stub>
    </aside>
  </v-layout-stub>
</v-container-stub>
`;
